File: anime_coloring.rb
# anime_coloring.rb — Ruby версия

COLORS = {
  'black' => 30, 'red' => 31, 'green' => 32, 'yellow' => 33,
  'blue' => 34, 'magenta' => 35, 'cyan' => 36, 'white' => 37,
  'gray' => 90, 'bright_red' => 91, 'bright_green' => 92,
  'bright_yellow' => 93, 'bright_blue' => 94, 'bright_magenta' => 95,
  'bright_cyan' => 96, 'bright_white' => 97
}

TEMPLATE = <<~TEXT

          {hair}██{hair}██{hair}██{hair}██{hair}██
        {hair}██{skin}      {skin}{hair}██
       {hair}██  {eye}██{hair}  {hair}██  {eye}██{hair}  {hair}██
       {hair}██  {eye}██{hair}  {hair}██  {eye}██{hair}  {hair}██
        {hair}██{skin}      {skin}{hair}██
         {hair}██{hair}██{hair}██{hair}██{hair}██{hair}██{hair}██{hair}
          {hair}██  {hair}██
         {hair}██{clothes}    {clothes}{hair}██
         {hair}██{clothes}    {clothes}{hair}██
          {hair}██{hair}██{hair}██{hair}██{hair}██
TEXT

def colorize(text, color_name)
  code = COLORS[color_name] || 37
  "\e[#{code}m#{text}\e[0m"
end

def generate_image(hair, eyes, skin, clothes)
  result = TEMPLATE.dup
  result.gsub!('{hair}', colorize('██', hair))
  result.gsub!('{eyes}', colorize('██', eyes))
  result.gsub!('{skin}', colorize('  ', skin))
  result.gsub!('{clothes}', colorize('██', clothes))
  result
end

def main
  hair = 'magenta'
  eyes = 'cyan'
  skin = 'yellow'
  clothes = 'blue'
  output = nil

  args = ARGV
  i = 0
  while i < args.size
    case args[i]
    when '--hair' then hair = args[i+1]; i += 2
    when '--eyes' then eyes = args[i+1]; i += 2
    when '--skin' then skin = args[i+1]; i += 2
    when '--clothes' then clothes = args[i+1]; i += 2
    when '--output' then output = args[i+1]; i += 2
    else i += 1
    end
  end

  puts '🎨 Anime Coloring (Ruby)'
  puts "Цвета: волосы=#{hair}, глаза=#{eyes}, кожа=#{skin}, одежда=#{clothes}"
  puts
  image = generate_image(hair, eyes, skin, clothes)
  puts image

  if output
    File.write(output, image + "\n")
    puts "💾 Сохранено в #{output}"
  end
end

main if __FILE__ == $0

File: anime_coloring.py


### 1. `anime_coloring.py` (Python)

```python
# anime_coloring.py — Python версия

import sys
import argparse

# Карта цветов ANSI
COLORS = {
    'black': 30, 'red': 31, 'green': 32, 'yellow': 33,
    'blue': 34, 'magenta': 35, 'cyan': 36, 'white': 37,
    'gray': 90, 'bright_red': 91, 'bright_green': 92,
    'bright_yellow': 93, 'bright_blue': 94, 'bright_magenta': 95,
    'bright_cyan': 96, 'bright_white': 97
}

# ASCII-арт персонажа с областями
# Формат: [метка] для замены цветом
TEMPLATE = """
          {hair}██{hair}██{hair}██{hair}██{hair}██
        {hair}██{skin}      {skin}{hair}██
       {hair}██  {eye}██{hair}  {hair}██  {eye}██{hair}  {hair}██
       {hair}██  {eye}██{hair}  {hair}██  {eye}██{hair}  {hair}██
        {hair}██{skin}      {skin}{hair}██
         {hair}██{hair}██{hair}██{hair}██{hair}██{hair}██{hair}██{hair}
          {hair}██  {hair}██
         {hair}██{clothes}    {clothes}{hair}██
         {hair}██{clothes}    {clothes}{hair}██
          {hair}██{hair}██{hair}██{hair}██{hair}██
"""

def colorize(text, color_name):
    """Обёртывает текст в ANSI-код цвета."""
    if color_name not in COLORS:
        color_name = 'white'
    code = COLORS[color_name]
    return f"\033[{code}m{text}\033[0m"

def generate_image(colors):
    """Заменяет метки в шаблоне на цветные символы."""
    result = TEMPLATE
    for key, value in colors.items():
        # Заменяем {key} на цветной текст с символом '█' (или пробелом для skin)
        if key == 'skin':
            replacement = colorize('  ', value)  # два пробела для кожи
        else:
            replacement = colorize('██', value)
        result = result.replace('{' + key + '}', replacement)
    # Убираем лишние пробелы, оставляя структуру
    lines = result.split('\n')
    # Обрезаем пробелы в начале каждой строки (сохраняя отступы)
    return '\n'.join(lines)

def main():
    parser = argparse.ArgumentParser(description='Anime Coloring')
    parser.add_argument('--hair', default='magenta', help='Цвет волос')
    parser.add_argument('--eyes', default='cyan', help='Цвет глаз')
    parser.add_argument('--skin', default='yellow', help='Цвет кожи')
    parser.add_argument('--clothes', default='blue', help='Цвет одежды')
    parser.add_argument('--output', '-o', help='Сохранить в файл')
    args = parser.parse_args()

    colors = {
        'hair': args.hair,
        'eyes': args.eyes,
        'skin': args.skin,
        'clothes': args.clothes
    }

    image = generate_image(colors)
    print("🎨 Anime Coloring (Python)")
    print("Цвета: волосы={}, глаза={}, кожа={}, одежда={}".format(
        args.hair, args.eyes, args.skin, args.clothes))
    print()
    print(image)

    if args.output:
        # Сохраняем с ANSI-кодами
        with open(args.output, 'w') as f:
            f.write(image + '\n')
        print(f"💾 Сохранено в {args.output}")

if __name__ == '__main__':
    main()
